doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -5,269 +5,297 @@ Reasoning with inductive types
 
-The tactics presented here specialize :tacn:`apply` (and
-:tacn:`eapply`) to the case of constructors of inductive types.
+The tactics presented here specialize :tacn:`apply` and
+:tacn:`eapply` to constructors of inductive types.
 
-.. tacn:: constructor @natural
-   :name: constructor
+.. tacn:: constructor {? @nat_or_var } {? with @bindings }
 
-   This tactic applies to a goal such that its conclusion is an inductive
-   type (say :g:`I`). The argument :token:`natural` must be less or equal to the
-   numbers of constructor(s) of :g:`I`. Let :n:`c__i` be the i-th
-   constructor of :g:`I`, then :g:`constructor i` is equivalent to
-   :n:`intros; apply c__i`.
+   First does :n:`repeat intro; hnf` on the goal.  If the result is an inductive
+   type, then apply the appropriate constructor(s), and otherwise fail.
+   If :n:`@nat_or_var` is specified and has the
+   value `i`, it uses :n:`apply c__i`, where :n:`c__i` is the i-th constructor
+   of :g:`I`.  If not specified, the tactic tries all the constructors,
+   which can result in more than one match (e.g. for `\\/`) in compound backtracking
+   tactics such as `constructor; ...` is used.
+
+   :n:`{? with @bindings }`
+     If specified, the :n:`apply` is done as :n:`apply … with @bindings`.
+
+     .. warning::
+
+        The terms in :token:`bindings` are checked in the context
+        where constructor is executed and not in the context where :tacn:`apply`
+        is executed (the introductions are not taken into account).
 
    .. exn:: Not an inductive product.
       :undocumented:
 
    .. exn:: Not enough constructors.
       :undocumented:
 
-   .. tacv:: constructor
-
-      This tries :g:`constructor 1` then :g:`constructor 2`, ..., then
-      :g:`constructor n` where ``n`` is the number of constructors of the head
-      of the goal.
-
-   .. tacv:: constructor @natural with @bindings
-
-      Let ``c`` be the i-th constructor of :g:`I`, then
-      :n:`constructor i with @bindings` is equivalent to
-      :n:`intros; apply c with @bindings`.
-
-      .. warning::
+   .. todo PR: maybe split, left, right etc should be at the left margin?
 
-         The terms in :token:`bindings` are checked in the context
-         where constructor is executed and not in the context where :tacn:`apply`
-         is executed (the introductions are not taken into account).
+   .. tacn:: split {? with @bindings }
 
-   .. tacv:: split {? with @bindings }
-      :name: split
+      Equivalent to :n:`constructor 1 {? with @bindings }` when the
+      conclusion is an inductive type with a single constructor. It is
+      typically used to split conjunctions in the conclusion such as `A /\\ B` into
+      two new goals `A` and `B`.
 
-      This applies only if :g:`I` has a single constructor. It is then
-      equivalent to :n:`constructor 1 {? with @bindings }`. It is
-      typically used in the case of a conjunction :math:`A \wedge B`.
+      .. tacn:: exists {*, @bindings }
```

Comment:
"a shared description" - do you mean to put these lines together and meld the text? Or something else?

```
.. tacn:: split {? with @bindings }
.. tacn:: exists {*, @bindings }
```

Reply:
I'm not seeing how to meld them in a helpful way.  I tweaked them slightly; `exists` was a bit incorrect.  I think they are correct now.  Can you elaborate on how you would meld them?

FWIW, `split` does a bit more than advertised:

```
Goal forall A B, A /\ B -> True.
split.   (* No more subgoals. *)
```

Reply:
> FWIW, split does a bit more than advertised:

This is because `True` has only one constructor with no arguments, hence `constructor 1` performs an implicit `intros` and then solves the goal.  Is this missing from the advertising?